assets/app/scripts/controllers/javaLink.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:JavaLinkController
+ * @description
+ * # ProjectController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .run(function(ProxyPod, BaseLink, HawtioExtension, $templateCache, $compile) {
+    var template =
+'<span ng-show="container.jolokiaUrl" title="View java details">' +
+'  <a ng-click="gotoContainerView(container, container.jolokiaUrl)" href="">' +
+'    <i class="fa fa-external-link"></i>' +
+'  </a>' +
+'</span>';
+    HawtioExtension.add('container-links', function ($scope) {
+      var pod = (UI.findParentWith($scope, 'pod') || {}).pod;
+      var container = $scope.container;
+      console.log("pod: ", pod, " container: ", container);
+      if (pod && container) {
+        var podId = pod.metadata.name;
+        var namespace = pod.metadata.namespace;
+        var ports = Core.pathGet(container, ['ports']);
+        if (ports && ports.length > 0) {
+          var jolokiaPort = _.find(ports, function(port) {
```

Comment:
Instead of `Core.pathGet(...)` and `if (ports && ports.length)`, I would just do 
```
var jolokiaPort = _.find((container.ports || []), function(port) {
  ...
```